src/core_ocean/shared/mpas_ocn_vmix_gotm.F

```diff
@@ -0,0 +1,520 @@
+      !
+      call mpas_pool_get_array(meshPool, 'maxLevelCell', maxLevelCell)
+      call mpas_pool_get_array(meshPool, 'bottomDepth', bottomDepth)
+
+      call mpas_pool_get_array(statePool, 'layerThickness', layerThickness, timeLevel)
+
+      !
+      ! set pointers for fields related ocean state
+      !
+      call mpas_pool_get_array(diagnosticsPool, 'BruntVaisalaFreqTop',BruntVaisalaFreqTop)
+
+      !
+      ! set pointers for fields related forcing at ocean surface
+      !
+      call mpas_pool_get_array(diagnosticsPool, 'surfaceFrictionVelocity', surfaceFrictionVelocity)
+
+      !
+      ! set pointer for z at the midpoint of vertical grid
+      !
+      call mpas_pool_get_array(diagnosticsPool, 'zMid', zMid)
+
+      !
+      ! set pointers for viscosity/diffusivity
+      !
+      call mpas_pool_get_array(diagnosticsPool, 'vertViscTopOfCell', vertViscTopOfCell)
+      call mpas_pool_get_array(diagnosticsPool, 'vertDiffTopOfCell', vertDiffTopOfCell)
+
+      ! set pointer for kinetic energy
+      call mpas_pool_get_array(diagnosticsPool, 'kineticEnergyCell', kineticEnergyCell)
+
+      !
+      ! set pointers for GOTM viscosity/diffusivity, TKE, rate of dissipation and turbulent length scale
+      ! (current and new state)
+      !
+      call mpas_pool_get_array(statePool, 'gotmVertViscTopOfCell', gotmVertViscTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmVertViscTopOfCell', gotmVertViscTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmVertDiffTopOfCell', gotmVertDiffTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmVertDiffTopOfCell', gotmVertDiffTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmTKETopOfCell', gotmTKETopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmTKETopOfCell', gotmTKETopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmDissTopOfCell', gotmDissTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmDissTopOfCell', gotmDissTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmLengthTopOfCell', gotmLengthTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmLengthTopOfCell', gotmLengthTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmKbTopOfCell', gotmKbTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmKbTopOfCell', gotmKbTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmEpsbTopOfCell', gotmEpsbTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmEpsbTopOfCell', gotmEpsbTopOfCellNew, 2)
+
+      nCells = nCellsArray( size(nCellsArray) )
+
+      !$omp do schedule(runtime)
+      do iCell = 1, nCells
+         vertViscTopOfCell(:, iCell) = 0.0_RKIND
+         vertDiffTopOfCell(:, iCell) = 0.0_RKIND
+      end do
+      !$omp end do
+
+      ! initialize GOTM arrays at the first step
+      ! MPAS-O (1:nlev+1) <-> GOTM (nlev:0)
+      if (first) then
+         !$omp do schedule(runtime)
```

Comment:
@qingli411 all our omp calls have changed format in https://github.com/MPAS-Dev/MPAS-Model/pull/513. Please search for all !$omp lines in your PR and change to this:
```
      !$omp parallel
      !$omp do schedule(runtime) &
      !$omp private(k, ki)
      do iCell = 1, nCells
...
```
for all cell, edge, and vertex loops. The private variables are any index or scalar value used within the loop.

Reply:
Note that the first index (here iCell) is not in the private line on purpose. That is because openMP knows to run thread-parallel on that first loop.

Reply:
Initialization of the GOTM state variables are now in the init subroutine (see 7b00c93).

I also changed the format for the OpenMP directives. But since my knowledge of OpenMP is very limited I'm not sure whether it work with OpenMP... Here GOTM has its own initialization step (called in the subroutine ` ocn_vmix_gotm_init`), which allocates memory for a list of 1D arrays (vertical profiles of TKE, length scales, viscosity etc). Then I loop over all the MPAS-O cells to call GOTM to update these variables for each cell (the GOTM state variables allocated in MPAS-O).